core/src/avm2/globals/flash/display3D/textures/Texture.as

```diff
@@ -1,11 +1,24 @@
 package flash.display3D.textures {	
     import flash.display.BitmapData;
+    import flash.events.Event;
     import flash.utils.ByteArray;
-    import __ruffle__.stub_method;
+    import flash.utils.setTimeout;
     
     public final class Texture extends TextureBase {
         public native function uploadFromBitmapData(source:BitmapData, miplevel:uint = 0):void;
         public native function uploadFromByteArray(data:ByteArray, byteArrayOffset:uint, miplevel:uint = 0):void;
-        public native function uploadCompressedTextureFromByteArray(data:ByteArray, byteArrayOffset:uint, async:Boolean = false):void;
+        public function uploadCompressedTextureFromByteArray(data:ByteArray, byteArrayOffset:uint, async:Boolean = false):void {
+            if (async) {
+                var self = this;
+                setTimeout(function() {
+                    self.uploadCompressedTextureFromByteArrayInternal(data, byteArrayOffset);
```

Comment:
Can you move this out of the `setTimeout`? I've seen Stage3D code that breaks when events are fired from within the method (the listener gets registered afterward).

Reply:
I don't see how this could break anything? Starling registers the event listeners after calling `uploadCompressedTextureFromByteArray`, so these need to be in a setTimeout.

Reply:
I mean, move just the `uploadCompressedTextureFromByteArrayInternal` call out of the `setTimeout`, and leave the `dispathEvent` call in the `setTimeout`

Reply:
I don't think the current implementation can break anything? The Adobe docs say that the function immediately returns and dispatches the event after `uploadCompressedTextureFromByteArrayInternal` has been called.

Reply:
Oh, I see what you mean. Yeah, this works fine